prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -530,11 +620,97 @@ def canonicalize_dataset_automatically(
             return canonicalized_dataset
 
+    def get_datasets_of_required_size(
+        self, datasets_info_dict: dict, prompt_spec: PromptSpec
+    ) -> datasets.DatasetDict:
+        """Combine multiple datasets to get the required size.
+
+        Args:
+            datasets_info_dict (dict): A list of dictionaries representing the datasets.
+            prompt_spec (PromptSpec): An object representing the prompt specification.
+
+        Returns:
+            transformed_dataset: The transformed dataset containing the required
+            number of points.
+
+        """
+        curr_datasets_size = 0
+        inputs = []
+        outputs = []
+        dataset_contributions = {}
+        number_of_chosen_datasets = 0
+        while (
+            curr_datasets_size < self.total_num_points_to_transform
+            and len(datasets_info_dict.keys()) > 0
+            and number_of_chosen_datasets <= self.max_datasets_to_choose
+        ):
+            dataset_name, config_name = self.rerank_datasets(
+                datasets_info_dict, prompt_spec
+            )
+            if dataset_name is None:
+                # If it couldn't find a relevant dataset from reranking
+                # (even after voting) stop trying to find more datasets.
+                return None
+            number_of_chosen_datasets += 1
+
+            if config_name is None:
+                del datasets_info_dict[dataset_name]
+                continue  # If it couldn't find a relevant config, delete the entire dataset. # noqa: E501
+
+            dataset_info = datasets_info_dict[dataset_name]["configs"][config_name]
+            canonicalized_dataset = self.canonicalize_dataset_automatically(
+                dataset_info,
+                prompt_spec,
+                self.total_num_points_to_transform - curr_datasets_size,
+            )
+            curr_datasets_size += len(canonicalized_dataset["train"]["input_col"])
+            inputs += canonicalized_dataset["train"]["input_col"]
+            outputs += canonicalized_dataset["train"]["output_col"]
+            dataset_contributions[f"{dataset_name}_{config_name}"] = len(
+                canonicalized_dataset["train"]["input_col"]
+            )
+
+            if len(datasets_info_dict[dataset_name]["configs"]) == 1:
+                del datasets_info_dict[dataset_name]
+            else:
+                del datasets_info_dict[dataset_name]["configs"][config_name]
+        logger.info(f"Chosen datasets: {dataset_contributions}")
+        transformed_dataset = self.make_dataset_from_samples(inputs, outputs)
+
+        return transformed_dataset
+
+    def create_dataset(
+        self, prompt_spec: PromptSpec, sorted_list: list[str]
+    ) -> datasets.DatasetDict | None:
+        """Creates the dataset based on the given prompt specification and sorted list.
+
+        Args:
+            prompt_spec (PromptSpec): The prompt specification.
+            sorted_list (list[str]): The sorted list.
```

Comment:
no need for redundant type hints
```suggestion
            prompt_spec: The prompt specification.
            sorted_list: The sorted list.
```